SU-HSDO/suhumsci
# STN-520: spacing bug fix for hb-views-divider class

# READY FOR REVIEW

## Summary
There is a spacing bug on the `hb-views-divider` utility wrapper class. It was providing margin to the bottom of the divider which was also collapsing within the `views-row` class and not providing spacing separation between the card and the divider line that is added.

Bug Card: [STN-520](https://sparkbox.atlassian.net/browse/STN-520)
The spacing issue on staging: https://sparkbox-sandbox-stage.stanford.edu/people-0

## Resolution
I removed the margin-bottom that was collapsing and added a padding-bottom that will give the card proper padding below the content but within the container (above the divider line). Adding the padding to the `hb-views-divider` utility class also ensures that this extra padding only shows on cards using that wrapper class with dividers and not on other `hb-card` lists.

It should look like this:
<img width="990" alt="Screen Shot 2020-08-21 at 2 56 13 PM" src="https://user-images.githubusercontent.com/9355698/90925024-93222000-e3be-11ea-9ba5-5c792dec7522.png">

I have browser checked this fix in Firefox, Chrome and Safari (current releases) and also IE11.

## Need Review By (Date)
10/24

## Urgency
medium

## Steps to Test
1. Run `npm test`
2. To review this fix, you will need to add the `hb-views-divider` class to the card view that you want to check.
 - Go to the People list page: http://sparkbox-sandbox.suhumsci.loc/people-0
- Click on the "edit view" contextual filter icon (pencil)
![image](https://user-images.githubusercontent.com/9355698/90923981-9fa57900-e3bc-11ea-87a7-be4fdcbe571d.png)

- Once in the View edit page, you'll want to click on the bottom right hand side of the options where it says "CSS class:", it will likely have no class present on your site, you'll want to click that and enter "hb-views-divider" into the modal window and click "apply". 
<img width="333" alt="Screen Shot 2020-08-21 at 2 43 44 PM" src="https://user-images.githubusercontent.com/9355698/90924879-49d1d080-e3be-11ea-8f05-48d8e3b3ec05.png">
<img width="386" alt="Screen Shot 2020-08-21 at 2 46 17 PM" src="https://user-images.githubusercontent.com/9355698/90924887-4c342a80-e3be-11ea-8dd7-742d6cbbe6fa.png">

- Once that is complete, you'll want to "Save" at the bottom of the page to save the View options and apply the class to your view.
3. You should see the divider with the spacing now applying properly and you can compare it to [staging](http://sparkbox-sandbox.suhumsci.loc/people-0), you'll want to set your theme appearance to Humsci Basic Traditional for direct comparison, however the divider fix should work on both Colorful themes and Traditional.
(For comparison without the edit button, comparing the pages in a incognito window for "logged out" state works best.)
4. If you'd like to test an additional view, http://sparkbox-sandbox.suhumsci.loc/qa-views has a view at the top that also is a good candidate for adding the utility class to for additional testing if needed.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
